### ISSUE TITLE
[OSX] Installer: only restart Finder on Systems <10.10

### ISSUE THREAD

**Issue by stonerl** [owncloud/client#3586]
### Expected behaviour

Since the native Sharing-API is used in v.2.0 when installed on Yosemite. Finder does not have to be restarted on such Systems.
### Actual behaviour

The Installer restarts Finder.
### Client configuration

Client version: 2.0.0.2622-nightly20150811

Operating system: OSX 10.10.4

**Comment by guruz** [comment]
Is that a problem?

**Comment by stonerl** [comment]
Well, it is not necessary. This would be the check:

```bash
OSXVERSION=$(sw_vers -productVersion | cut -d . -f 2)

if (($OSXVERSION < 10)); then
	sudo killall Finder
fi
```

**Comment by guruz** [comment]
Let's do this for 2.1 :)
For 2.0, people might still have the legacy Finder extension (from 1.8.x or before) installed.. it's better to kill it off from memory.. and Finder is auto-restarting.

**Comment by guruz** [comment]
With regards to https://github.com/owncloud/client/issues/4650 I'm thinking maybe the ```killall Finder``` should stay in and be in ```admin/osx/post_install.sh.cmake```

**Comment by stonerl** [comment]
Not really. OSX does load this Extension dynamically. There is no need to `killall Finder`, e.g. Dropbox also utilizes the Finder integration and they do not kill Finder.

1. Open `System Preferences --> Extensions` there you can see the ownCloud-Extension. 
2. Leave the Window open and move `owncloud.app` into the Trash.
3. Wait a second and the extensions disappears from the System Preferences.
4. Move `owncloud.app` from Trash back into `/Applications`
5. Wait some seconds and the Extensions is back in the System Preferences

**Comment by guruz** [comment]
@stonerl is most probably right. I had fixed the issue $4650 in another way which had similar effects ... https://github.com/owncloud/client/commit/6a9655aab6b0b06dc4085dcb01ff7ef4131100ee :)

**Comment by guruz** [comment]
@jturcotte Hmmhm maybe still better to do this also with latest overlay versions so we try harder to get Finder to display our icons?

**Comment by jturcotte** [comment]
We talked about removing the <10.10 Finder plugin altogether and eventually even replace the .pkg with a .dmg (which would, from my understanding, remove the ability to even run this kind of script). This will probably be how this get fixed.

**Comment by jturcotte** [comment]
Fixed by 3b20684dc791d1c8c6034e95a0e4b65d722bcfc0
